domain/src/test/java/org/oppia/android/domain/exploration/ExplorationProgressControllerTest.kt

```diff
@@ -209,6 +221,77 @@ class ExplorationProgressControllerTest {
+    oppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    val currentTime = oppiaClock.getCurrentTimeMs()
+    oppiaClock.setCurrentTimeMs(currentTime)
+    startPlayingNewExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2)
+    val ephemeralState = waitForGetCurrentStateSuccessfulLoad()
+    assertThat(ephemeralState.showContinueButtonAnimation).isEqualTo(true)
+    assertThat(ephemeralState.continueButtonAnimationTimestampMs).isEqualTo(
+      currentTime + TimeUnit.SECONDS.toMillis(
+        45
+      )
+    )
+  }
+
+  @Test
+  fun testEphemeralState_moveToNextState_shouldIndicateNoButtonAnimation() {
+    startPlayingNewExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2)
+    playThroughPrototypeState1AndMoveToNextState()
+    val ephemeralState = waitForGetCurrentStateSuccessfulLoad()
+    assertThat(ephemeralState.showContinueButtonAnimation).isEqualTo(false)
+  }
+
+  @Test
+  fun testEphemeralState_profile1ClicksContinue_switchToProfile2_shouldIndicateButtonAnimation() {
+    oppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    val profileId2 = ProfileId.newBuilder().setInternalId(1).build()
+    startPlayingNewExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2)
+    playThroughPrototypeState1AndMoveToNextState()
+    endExploration()
+    val currentTime = oppiaClock.getCurrentTimeMs()
+    oppiaClock.setCurrentTimeMs(currentTime)
+    startPlayingNewExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2, profileId2)
+    val ephemeralState = waitForGetCurrentStateSuccessfulLoad()
+    assertThat(ephemeralState.showContinueButtonAnimation).isEqualTo(true)
+    assertThat(ephemeralState.continueButtonAnimationTimestampMs).isEqualTo(
+      currentTime + TimeUnit.SECONDS.toMillis(
+        45
+      )
+    )
+  }
+
+  @Test
+  fun testEphemeralState_startExp_clicksContinue_reEnterExp_shouldIndicateNoButtonAnimation() {
+    startPlayingNewExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2)
+    playThroughPrototypeState1AndMoveToNextState()
+    val checkPoint = retrieveExplorationCheckpoint(TEST_EXPLORATION_ID_2)
+    endExploration()
+    resumeExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2, checkPoint)
+    val ephemeralState = waitForGetCurrentStateSuccessfulLoad()
+    assertThat(ephemeralState.showContinueButtonAnimation).isEqualTo(false)
+  }
+
+  @Test
+  fun testEphemeralState_startExp_seesAnimation_reEnterExploration_shouldIndicateButtonAnimation() {
+    oppiaClock.setFakeTimeMode(FakeOppiaClock.FakeTimeMode.MODE_FIXED_FAKE_TIME)
+    startPlayingNewExploration(TEST_TOPIC_ID_0, TEST_STORY_ID_0, TEST_EXPLORATION_ID_2)
+    testCoroutineDispatchers.advanceTimeBy(TimeUnit.SECONDS.toMillis(45))
+    endExploration()
+    val currentTime = oppiaClock.getCurrentTimeMs()
+    oppiaClock.setCurrentTimeMs(currentTime)
```

Comment:
Here & elsewhere in the code above: why is this needed? Can't we just fetch the time? Not sure I understand why we need to set it back to the same time it started at.

Reply:
done -- removed